frontend/system/src/Entities/ListView.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from '@common/intl';
+import {
+  AppBarContentPortal,
+  DataTable,
+  NothingHere,
+  TableProvider,
+  createTableStore,
+  useColumns,
+  SearchToolbar,
+  ToggleButton,
+} from '@common/ui';
+import { useBreadcrumbs, useQueryParamsState } from '@common/hooks';
+import { useEntities } from '../api';
+import { ToggleButtonGroup } from '@mui/material';
+
+export const ListView = () => {
+  const t = useTranslation('system');
+  const { setSuffix } = useBreadcrumbs();
+
+  useEffect(() => {
+    setSuffix('Browse');
+  }, []);
+
+  const { filter, queryParams, updatePaginationQuery, updateSortQuery } =
+    useQueryParamsState({
+      initialSort: { key: 'description', dir: 'asc' },
+    });
+
+  const { sortBy, page, offset, first } = queryParams;
+
+  const columns = useColumns(
+    [
+      { key: 'code', label: 'label.code', width: 200, sortable: false },
+      { key: 'description', label: 'label.description', width: 1000 },
+      { key: 'type', label: 'label.type', sortable: false, width: 200 },
+    ],
+    { sortBy: sortBy, onChangeSortBy: updateSortQuery },
+    [sortBy, updateSortQuery]
+  );
+
+  const [categories, setCategories] = useState<string[]>(['drug']);
+
+  const searchFilter = filter.filterBy?.['search'];
+  const search = typeof searchFilter === 'string' ? searchFilter : '';
+
+  const { data, isError, isLoading } = useEntities({
+    filter: {
+      categories,
+      description: search,
+      orderBy: {
+        field: sortBy.key,
+        descending: sortBy.isDesc,
+      },
+    },
+    first,
+    offset,
+  });
+
+  const updateCategories = (category: string) => {
```

Comment:
```suggestion
  const toggleCategory = (category: string) => {
```
Not a major, just had to check the code to understand what it did

Reply:
Yeah nice catch, [0c0918d](https://github.com/msupply-foundation/unified-codes/pull/434/commits/0c0918d9debe2a13a159897173927a66934a0c1d)